src/libs/actions/Report.js

```diff
@@ -307,6 +307,7 @@ function fetchChatReportsByIDs(chatList) {
                 simplifiedReports[reportKey].iouReportID = iouReportObject.reportID;
                 simplifiedReports[reportKey].hasOutstandingIOU = iouReportObject.stateNum === 1
                     && iouReportObject.total !== 0;
+                Onyx.merge(iouReportKey, iouReportObject);
```

Comment:
Maybe I'm missing something, but shouldn't this be redundant? As the whole simplifiedReports collection is merged to Onyx in line 316/317?

Is mergeCollection not triggering an update? Because if so that sounds like an issue @barun1997 is currently facing too.  I'll take a look into this today.

Reply:
I have found that Onyx.mergeCollection doesn't trigger update on individual key subscribers and it was causing delay on updating outstanding amount. Adding Onyx.merge here fixed that issue.

Reply:
Yup, this sounds exactly like the issue I was facing. But at the same time, I don't think Onyx.merging each individual key is a good fix to the issue since it adds too many renders if I am not wrong, @tugbadogan? Do give me feedback if you've any suggestions so I could implement the same in my PR as well.

Reply:
Okay, thanks for confirming. I'll get back to you

Reply:
This was a bug in Onyx, which [has just been fixed](https://github.com/Expensify/react-native-onyx/pull/57). We've just [bumped](https://github.com/Expensify/Expensify.cash/pull/2200) the Onyx version in Expensify.cash, so you should be able to resolve your issue after merging master.

Reply:
👍 Thanks for confirming @Julesssss 

I have a question about rebasing to master. It looks like Github requires force push after rebase and in Expensifh.cash contributor guidelines it's recommended not to force push since it ignored previous reviews. 

Do you know if it's possible to rebase and update the PR without force push? or is it OK to force push after rebase?

Reply:
I believe in this case it is required to force push, so it is OK. We will discuss internally the changes required to the contributor guidelines for future cases of this. Thank you for double checking first 👍

Reply:
Hey @tugbadogan are you all good here now?

Reply:
I didn't have a change to look at it today. I'm planning to test it tomorrow. 👍

Reply:
I've tested with most recent Onyx version and it seems to be working but I have discovered another problem and created a separate issue for it  https://github.com/Expensify/Expensify.cash/issues/2225

This issue is causing preview box to disappear when it's hovered with mouse. Since `Onyx.mergeCollection` doesn't update report object properly, we don't have `hasOutstandingIOU` value available in this component.

@Julesssss is there any chance your test didn't work as expected due to this issue?


https://user-images.githubusercontent.com/3853003/113597295-6829cb00-9633-11eb-87a9-fc54b29292ad.mov

Reply:
Continuing discussion of bug found in linked issue